django/utils/dateparse.py

```diff
@@ -87,12 +90,15 @@ def parse_time(value):
     Return None if the input isn't well formatted, in particular if it
     contains an offset.
     """
-    match = time_re.match(value)
-    if match:
-        kw = match.groupdict()
-        kw['microsecond'] = kw['microsecond'] and kw['microsecond'].ljust(6, '0')
-        kw = {k: int(v) for k, v in kw.items() if v is not None}
-        return datetime.time(**kw)
+    try:
+        return datetime.time.fromisoformat(value).replace(tzinfo=None)
```

Comment:
I'm happy enough that we do this, but it's perhaps worth having a comment to explain why.